File: P1_Two_Sum/CPlusPlus/other_imp1/main.cpp
#include <iostream>
#include <cstdlib>
#include <cstring>
#include <climits>
#include <vector>
#include <solution.h>

int main(){
    //Declare final solution container.
    std::vector<int> final_answer;

    //The input testing arguments.
    int target = 12;
    int myints[] = {2, 3, 3, 9};
    std::vector<int> nums(myints, myints+sizeof(myints)/sizeof(int));

    //Calculation body
    Solution sol_obj;
    final_answer = sol_obj.twoSum(nums, target);

    //Print the result.
    for (auto it = final_answer.begin(), end = final_answer.end(); it != end; ++it){
        if(it==final_answer.begin()){
            printf("answer = [%d", *it);
        }else if(it==end-1){
            printf(", %d]\n", *it);
        }else{
            printf(", %d", *it);
        }
    }

    return EXIT_SUCCESS;
}

File: P1_Two_Sum/Python/other_imp1/main.py
#! /usr/bin/env python3

from solution import Solution

def main():
    sol = Solution()
    nums = [3, 2, 4]
    target = 6

    ans = sol.twoSum(nums, target)
    print(f"nums = {nums}, target = {target}")
    print(f"ans = {ans}")

#---------------Execution---------------#
if __name__ == '__main__':
    main()
